lib/column/variant_test.go

```diff
@@ -0,0 +1,91 @@
+package column
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestColVariant_parse(t *testing.T) {
+	cases := []struct {
+		typ           Type
+		expectedTypes []Type
+	}{
+		{typ: "Variant(Int64)", expectedTypes: []Type{"Int64"}},
+		{typ: "Variant(Int64, String)", expectedTypes: []Type{"Int64", "String"}},
+		{typ: "Variant(Array(String), Int64, String)", expectedTypes: []Type{"Array(String)", "Int64", "String"}},
+		{typ: "Variant(Array(Map(String, String)), Map(String, Int64))", expectedTypes: []Type{"Array(Map(String, String))", "Map(String, Int64)"}},
+		{typ: "Variant(Array(Map(String, Tuple(a String, b Int64))), Map(String, Int64))", expectedTypes: []Type{"Array(Map(String, Tuple(a String, b Int64)))", "Map(String, Int64)"}},
+	}
+
+	for i, c := range cases {
+		col, err := (&Variant{name: "vt"}).parse(c.typ, nil)
+		if err != nil {
+			t.Fatalf("case index %d failed to parse Variant column: %s", i, err)
+		}
+
+		require.Equal(t, "vt", col.Name())
+		require.Equal(t, c.typ, col.chType)
+		require.Equal(t, len(c.expectedTypes), len(col.columns))
+		require.Equal(t, len(c.expectedTypes), len(col.columnTypeIndex))
+
+		for j, subCol := range col.columns {
+			expectedType := c.expectedTypes[j]
+			actualType := subCol.Type()
+			if actualType != expectedType {
+				t.Fatalf("case index %d Variant type index %d column type does not match: expected: \"%s\" actual: \"%s\"", i, j, expectedType, actualType)
+			}
```

Comment:
why not testify assert function?

Reply:
In this case I wanted to be able to specify an index via `%d`. I can check the testify asserts to see if there's a function for this